From c72e0914f9193d9c27700d18b51bccce9fd1a7d5 Mon Sep 17 00:00:00 2001
From: Mahadev Konar <mahadev@apache.org>
Date: Mon, 17 Dec 2012 07:12:12 +0000
Subject: [PATCH 37/94] ZOOKEEPER-1578.
 org.apache.zookeeper.server.quorum.Zab1_0Test failed
 due to  hard code with 33556 port. (Li Ping via
 mahadev)

git-svn-id: https://svn.apache.org/repos/asf/zookeeper/branches/branch-3.4@1422771 13f79535-47bb-0310-9956-ffa450edef68

Conflicts:
	CHANGES.txt
---
 CHANGES.txt                                        |    3 +++
 .../apache/zookeeper/server/quorum/Zab1_0Test.java |    3 ++-
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/CHANGES.txt b/CHANGES.txt
index b199702..2193739 100644
--- a/CHANGES.txt
+++ b/CHANGES.txt
@@ -19,6 +19,9 @@ BUGFIXES:
   ZOOKEEPER-1478. Small bug in QuorumTest.testFollowersStartAfterLeader( )
   (Alexander Shraer via fpj, breed, phunt)
 
+  ZOOKEEPER-1578. org.apache.zookeeper.server.quorum.Zab1_0Test failed due to
+  hard code with 33556 port. (Li Ping via mahadev)
+
 IMPROVEMENTS:
 
 
diff --git a/src/java/test/org/apache/zookeeper/server/quorum/Zab1_0Test.java b/src/java/test/org/apache/zookeeper/server/quorum/Zab1_0Test.java
index 91bd7fb..a35e2bd 100644
--- a/src/java/test/org/apache/zookeeper/server/quorum/Zab1_0Test.java
+++ b/src/java/test/org/apache/zookeeper/server/quorum/Zab1_0Test.java
@@ -36,6 +36,7 @@ import org.apache.jute.BinaryInputArchive;
 import org.apache.jute.BinaryOutputArchive;
 import org.apache.jute.InputArchive;
 import org.apache.jute.OutputArchive;
+import org.apache.zookeeper.PortAssignment;
 import org.apache.zookeeper.WatchedEvent;
 import org.apache.zookeeper.Watcher;
 import org.apache.zookeeper.Watcher.Event.EventType;
@@ -826,7 +827,7 @@ public class Zab1_0Test {
         peer.setTxnFactory(logFactory);
         Field addrField = peer.getClass().getDeclaredField("myQuorumAddr");
         addrField.setAccessible(true);
-        addrField.set(peer, new InetSocketAddress(33556));
+        addrField.set(peer, new InetSocketAddress(PortAssignment.unique()));
         ZKDatabase zkDb = new ZKDatabase(logFactory);
         LeaderZooKeeperServer zk = new LeaderZooKeeperServer(logFactory, peer, new ZooKeeperServer.BasicDataTreeBuilder(), zkDb);
         return zk;
-- 
1.7.9.5

